plugin/src/main/cpp/include/extensions/openxr_meta_colocation_discovery.h

```diff
@@ -0,0 +1,120 @@
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#pragma once
+
+#include <openxr/openxr.h>
+#include <godot_cpp/classes/open_xr_extension_wrapper_extension.hpp>
+#include <godot_cpp/templates/hash_map.hpp>
+#include <map>
+
+#include "util.h"
+
+using namespace godot;
+
+// Wrapper for the XR_META_colocation_discovery extension.
+class OpenXRMetaColocationDiscovery : public OpenXRExtensionWrapperExtension {
+	GDCLASS(OpenXRMetaColocationDiscovery, OpenXRExtensionWrapperExtension);
+
+public:
+	static OpenXRMetaColocationDiscovery *get_singleton();
+
+	OpenXRMetaColocationDiscovery();
+	~OpenXRMetaColocationDiscovery();
+
+	godot::Dictionary _get_requested_extensions() override;
+
+	void _on_instance_created(uint64_t instance) override;
+	uint64_t _set_system_properties_and_get_next_pointer(void *p_next_pointer) override;
+	void _on_session_created(uint64_t session) override;
+	void _on_instance_destroyed() override;
+	bool _on_event_polled(const void *event) override;
+
+	bool is_colocation_discovery_supported() const;
+
+	bool start_advertisement(const PackedByteArray &p_buffer);
+	bool stop_advertisement();
+	bool start_discovery();
+	bool stop_discovery();
+
+protected:
+	static void _bind_methods();
+
+private:
+	EXT_PROTO_XRRESULT_FUNC3(xrStartColocationAdvertisementMETA,
+			(XrSession), session,
+			(const XrColocationAdvertisementStartInfoMETA *), info,
+			(XrAsyncRequestIdFB *), advertisementRequestId)
+
+	EXT_PROTO_XRRESULT_FUNC3(xrStopColocationAdvertisementMETA,
+			(XrSession), session,
+			(const XrColocationAdvertisementStopInfoMETA *), info,
+			(XrAsyncRequestIdFB *), requestId)
+
+	EXT_PROTO_XRRESULT_FUNC3(xrStartColocationDiscoveryMETA,
+			(XrSession), session,
+			(const XrColocationDiscoveryStartInfoMETA *), info,
+			(XrAsyncRequestIdFB *), discoveryRequestId)
+
+	EXT_PROTO_XRRESULT_FUNC3(xrStopColocationDiscoveryMETA,
+			(XrSession), session,
+			(const XrColocationDiscoveryStopInfoMETA *), info,
+			(XrAsyncRequestIdFB *), requestId)
+
+	void cleanup();
+
+	void on_start_advertisement_complete(const XrEventDataStartColocationAdvertisementCompleteMETA *event);
+	void on_stop_advertisement_complete(const XrEventDataStopColocationAdvertisementCompleteMETA *event);
+	void on_advertisement_complete(const XrEventDataColocationAdvertisementCompleteMETA *event);
+	void on_start_discovery_complete(const XrEventDataStartColocationDiscoveryCompleteMETA *event);
+	void on_discovery_result(const XrEventDataColocationDiscoveryResultMETA *event);
+	void on_discovery_complete(const XrEventDataColocationDiscoveryCompleteMETA *event);
+	void on_stop_discovery_complete(const XrEventDataStopColocationDiscoveryCompleteMETA *event);
+
+	static OpenXRMetaColocationDiscovery *singleton;
+
+	std::map<godot::String, bool *> request_extensions;
+	bool meta_colocation_discovery_ext = false;
+
+	XrSession openxr_session = XR_NULL_HANDLE;
```

Comment:
You can use [SESSION](https://github.com/GodotVR/godot_openxr_vendors/blob/2dbc90fc788d1cd16ed9141ff8ec6942d1696c39/plugin/src/main/cpp/include/util.h#L126) to get access to the session.

Reply:
Fixed